packages/cdk8s-plus-17/src/service.ts

```diff
@@ -232,6 +244,14 @@ export class Service extends Resource {
       throw new Error('A service must be configured with a port');
     }
 
+    if (this.type === ServiceType.EXTERNAL_NAME && this.externalName === undefined) {
+      throw new Error('A service with type EXTERNAL_NAME requires an externalName prop');
+    }
+
+    if (this.externalName !== undefined && this.type !== ServiceType.EXTERNAL_NAME) {
```

Comment:
I'd say that if `type` is undefined but `externalName` is defined then we can safely imply that type is `EXTERNAL_NAME`. Only if type is explicitly set to something else then we want to fail.

Reply:
This is also what I proposed in the previous comment

Reply:
@eladb - how do you feel about checking to see if externalName is set in the props in the constructor.  If it is, we force the type to EXTERNAL_NAME.  Since the only way set the type, we should be safe without the check in `_toKube`

Something like
```
    if (props.externalName !== undefined) {
      this.type = ServiceType.EXTERNAL_NAME;
    } else {
      this.type = props.type ?? ServiceType.CLUSTER_IP;
    }
```